internal/confmap/provider/eecprovider/provider.go

```diff
@@ -143,7 +144,7 @@ func (p *provider) Retrieve(ctx context.Context, uri string, watcherFunc confmap
 			return nil, err
 		}
 		if cfg.authToken != "" {
-			req.Header.Add(ApiTokenHeader, cfg.authToken)
+			req.Header.Add(AuthHeaderKey, ApiTokenPrefix + cfg.authToken)
```

Comment:
Should we use a formatting string here maybe? Change above to: 
```golang
const ApiTokenPrefixFormat = "Api-Token %s"
```
and then here 
```suggestion
			req.Header.Add(AuthHeaderKey, fmt.Sprintf(ApiTokenPrefixFormat, cfg.authToken))
```

Reply:
That would work too. @evan-bradley what's more idiomatic in Go?

Reply:
Neither from what I've gathered, for something simple like this it's up to the author.

Reply:
I have a slight preference for `fmt.Sprintf` for readability, though raw string concatenation is probably infinitesimally faster. I'm okay either way.

Reply:
Updated to use the format and changed the tests to use their own literals independent from the implementation so they're decoupled.